spec/unit/classes/globals_spec.rb

```diff
@@ -43,21 +43,21 @@
         is_expected.to contain_class("postgresql::globals")
       end
     end
-    
+
     describe 'manage_package_repo on RHEL => true' do
       let(:params) do
         {
           :manage_package_repo => true,
           :repo_proxy          => 'http://proxy-server:8080',
         }
       end
-      
+
       it 'should pull in class postgresql::repo' do
         is_expected.to contain_class("postgresql::repo")
       end
 
       it do
-        should contain_yumrepo('yum.postgresql.org').with(
+        should contain_yumrepo('download.postgresql.org').with(
```

Comment:
So, adding a test that passes in a different global for base_repourl, and checks to see if the repo_url is indeed different will prove your solution properly.